spec/models/queen_spec.rb

```diff
@@ -0,0 +1,92 @@
+      white_queen = game.pieces.find_by(rank: 1, file: 1)
+      white_queen2 = game.pieces.find_by(rank: 1, file: 3)
+      black_queen = game.pieces.find_by(rank: 8, file: 1)
+      black_queen2 = game.pieces.find_by(rank: 8, file: 8)
+      black_queen3 = game.pieces.find_by(rank: 8, file: 3)
+      expect(white_queen.valid_move?(1, 2)).to eq(false)
+      expect(white_queen.valid_move?(2, 1)).to eq(false)
+      expect(white_queen2.valid_move?(2, 4)).to eq(false)
+      expect(white_queen2.valid_move?(2, 2)).to eq(false)
+      expect(black_queen.valid_move?(8, 2)).to eq(false)
+      expect(black_queen.valid_move?(7, 1)).to eq(false)
+      expect(black_queen2.valid_move?(8, 7)).to eq(false)
+      expect(black_queen3.valid_move?(7, 4)).to eq(false)
+      expect(black_queen3.valid_move?(7, 2)).to eq(false)
+    end
+
+    it "should properly evaluate unobstructed, well-formed moves that are not allowed for a Queen" do
+      white_queen = game.pieces.create(type: "Queen", rank: 4, file: 4, color: "white")
+      white_queen2 = game.pieces.create(type: "Queen", rank: 4, file: 5, color: "white")
+      expect(white_queen.valid_move?(3, 2)).to eq(false)
+      expect(white_queen.valid_move?(3, 6)).to eq(false)
+      expect(white_queen.valid_move?(5, 2)).to eq(false)
+      expect(white_queen.valid_move?(5, 6)).to eq(false)
+      expect(white_queen2.valid_move?(5, 7)).to eq(false)
+      expect(white_queen2.valid_move?(5, 3)).to eq(false)
+      expect(white_queen2.valid_move?(6, 6)).to eq(false)
+      expect(white_queen2.valid_move?(6, 4)).to eq(false)
+      expect(white_queen2.valid_move?(3, 7)).to eq(false)
+      expect(white_queen2.valid_move?(3, 3)).to eq(false)
+      expect(white_queen2.valid_move?(2, 6)).to eq(false)
+      expect(white_queen2.valid_move?(2, 4)).to eq(false)
+    end
+
+    it "should properly evaluate valid moves" do
+      white_queen = game.pieces.create(type: "Queen", rank: 4, file: 4, color: "white")
+      white_queen2 = game.pieces.create(type: "Queen", rank: 6, file: 5, color: "white")
+      game.pieces.where(rank: 2).destroy_all
+      expect(white_queen.valid_move?(4, 3)).to eq(true)
+      expect(white_queen.valid_move?(4, 2)).to eq(true)
+      expect(white_queen.valid_move?(4, 5)).to eq(true)
+      expect(white_queen.valid_move?(4, 6)).to eq(true)
+      expect(white_queen.valid_move?(3, 4)).to eq(true)
+      expect(white_queen.valid_move?(2, 4)).to eq(true)
+      expect(white_queen.valid_move?(5, 4)).to eq(true)
+      expect(white_queen.valid_move?(6, 4)).to eq(true)
+      expect(white_queen2.valid_move?(5, 4)).to eq(true)
+      expect(white_queen2.valid_move?(5, 6)).to eq(true)
+      expect(white_queen2.valid_move?(7, 4)).to eq(true)
+      expect(white_queen2.valid_move?(7, 6)).to eq(true)
+      expect(white_queen2.valid_move?(3, 2)).to eq(true)
+      expect(white_queen2.valid_move?(4, 7)).to eq(true)
+    end
```

Comment:
Could you add a couple cases here that cover diagonal moves with increasing rank?